code/site/components/com_pages/data/object.php

```diff
@@ -25,17 +25,26 @@ public function slice($offset, $length = NULL)
         return new self($data);
     }
 
-    public function flatten()
+    public function flatten($property = null)
```

Comment:
@johanjanssens @ercanozkaya This feels dirty as pointed out by the turk, API wise it's not clean or clear. Perhaps ->flatten(true/false) to store the key value as a parent/folder property would work better instead.

Thoughts?

Reply:
Made the change here, much better IMO => https://github.com/joomlatools/joomlatools-pages/pull/52/commits/c6729e4c897496e70b9a1129a64fc68ae54bbeea

Reply:
@amazeika I would be careful with this approach, if a data object has a folder property we are overwriting it. 

What you are trying to do makes sense, but maybe we should expose this as a method on the object. Something like getFolder(), and store it in the object properties and not in the object data. That would prevent any conflict and we could make it so that this works all the time, not only when flattening? To do that we would need to make the data registry smarter to set this in the object when it's created.